Feature 500 v3.1.0 beta2

bikegeek opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for these changes:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**

- [ ] Do these changes include sufficient testing updates? **[Yes or No]**

- [ ] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Do these changes introduce new SonarQube findings? **[Yes or No]**</br>
If **yes**, please describe:

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METplotpy-X.Y.Z Development** project for official releases
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

bikegeek committed:
Updates for beta2 release)

bikegeek committed:
Update release-notes.rst

update the release date

bikegeek committed:
Update conf.py

update the date

jprestop commented:
Hi Minna.  I'm wondering if you missed my message in Slack:

> I wanted to let you know I updated the date for the METplotpy release to be 3/31 in the conf.py file and in the release notes in the develop branch since those changes had already been merged into develop.  I also noticed a typo in the release notes so I fixed that and then I modified some more of the release notes (also in develop) to make it consistent with METcalcpy and METdataio.  I hope you don't mind. 

So, that is likely why you're seeing conflicts.  I don't think you need any changes here.  Everything should be current in develop - unless you updated something.

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METplotpy into feature_500_v3.1.0-beta2

bikegeek committed:
Fix incorrect links

bikegeek commented:
I am fixing these, I updated the date but then had a bunch of conflicts I
needed to fix.  I am doing things incrementally.  Please DO NOT MAKE any
more changes, this creates more work for me.

On Mon, Mar 31, 2025 at 10:17 AM Julie Prestopnik ***@***.***>
wrote:

> Hi Minna. I'm wondering if you missed my message in Slack:
>
> I wanted to let you know I updated the date for the METplotpy release to
> be 3/31 in the conf.py file and in the release notes in the develop branch
> since those changes had already been merged into develop. I also noticed a
> typo in the release notes so I fixed that and then I modified some more of
> the release notes (also in develop) to make it consistent with METcalcpy
> and METdataio. I hope you don't mind.
>
> So, that is likely why you're seeing conflicts. I don't think you need any
> changes here. Everything should be current in develop - unless you updated
> something.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplotpy/pull/502#issuecomment-2766735499>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHRMVQG6MZMC4TSUVZL2XFTCTAVCNFSM6AAAAAB2EVCATSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRWG4ZTKNBZHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: jprestop]*jprestop* left a comment (dtcenter/METplotpy#502)
> <https://github.com/dtcenter/METplotpy/pull/502#issuecomment-2766735499>
>
> Hi Minna. I'm wondering if you missed my message in Slack:
>
> I wanted to let you know I updated the date for the METplotpy release to
> be 3/31 in the conf.py file and in the release notes in the develop branch
> since those changes had already been merged into develop. I also noticed a
> typo in the release notes so I fixed that and then I modified some more of
> the release notes (also in develop) to make it consistent with METcalcpy
> and METdataio. I hope you don't mind.
>
> So, that is likely why you're seeing conflicts. I don't think you need any
> changes here. Everything should be current in develop - unless you updated
> something.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplotpy/pull/502#issuecomment-2766735499>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHRMVQG6MZMC4TSUVZL2XFTCTAVCNFSM6AAAAAB2EVCATSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRWG4ZTKNBZHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

bikegeek committed:
Add back the New Plots and Enhancements drop downs

bikegeek committed:
Fix issue link for infrastructure issue #488 (replace METdataio with METplotpy and fix the issue number)

jprestop commented on docs/Users_Guide/release-notes.rst:
```diff
@@ -7,11 +7,16 @@ describes the bugfix, enhancement, or new feature:
 
-METplotpy Release Notes
-=======================
+METplotpy Version 3.1.0-beta1 release notes (20250331)
```
```suggestion
METplotpy Version 3.1.0-beta2 release notes (20250331)
```

jprestop commented on docs/Users_Guide/release-notes.rst:
```diff
@@ -22,23 +27,35 @@ METplotpy Version 3.1.0-beta2 release notes (20250331)
 .. dropdown:: Repository, build, and test
 
-   * Update infrastructure to reflect move to developing with Python 3.12 (`#469 <https://github.com/dtcenter/METdataio/pull/469>`_)
+   * Update infrastructure to reflect move to developing with Python 3.12 (`#488 <https://github.com/dtcenter/METplotpy/pull/488>`_)
```
```suggestion
   * Update infrastructure to reflect move to developing with Python 3.12 (`#496 <https://github.com/dtcenter/METplotpy/pull/496>`_)
```

jprestop commented on docs/Users_Guide/release-notes.rst:
```diff
@@ -7,11 +7,16 @@ describes the bugfix, enhancement, or new feature:
 
-METplotpy Version 3.1.0-beta2 release notes (20250331)
-------------------------------------------------------
+.. dropdown:: New Plots
```
Suggest removing any with "None" to better match the other METplus components. In Slack on 3/25, I sent the following note: 

> As you know, I always put in a title "Bugfixes", "Enhancements", etc, even if there are none and I put "None" underneath.  I see in the MET and METplus repos they just don't put the title "Bugfixes", "Enhancements", etc, if that category had no changes.  I guess I'll start doing the same for consistency...

jprestop commented on docs/Users_Guide/release-notes.rst:
```diff
@@ -22,23 +27,35 @@ METplotpy Version 3.1.0-beta2 release notes (20250331)
    * Plotly line plots are plotting confidence limit bars at zero with zero length error bars (`#495 <https://github.com/dtcenter/METplotpy/issues/495>`_)
 
```
These additional bugfixes used to be listed here, but I see they are no longer.  Did you mean to include them?
```
   * Series names in legend don't get ordered correctly  (`#347 <https://github.com/dtcenter/METplotpy/issues/347>`_)
   * Support various formats of show_legend values (`#482 <https://github.com/dtcenter/METplotpy/issues/482>`_)
   * Reliability Diagram show_legend setting should work with True/False rather than 1/0 values; see bugfix #482(`#455 <https://github.com/dtcenter/METplotpy/issues/455>`_)
   * Better handling of determining min and max for confidence limits when data contains NaN values(`#494 <https://github.com/dtcenter/METplotpy/issues/494>`_)
```

bikegeek replied:
they are included- you are probably looking at an earlier version that I was working on while resolving conflicts

jprestop commented on docs/Users_Guide/release-notes.rst:
```diff
@@ -22,23 +27,35 @@ METplotpy Version 3.1.0-beta2 release notes (20250331)
 METplotpy Version 3.1.0-beta1 release notes (20250123)
 ------------------------------------------------------
 
+.. dropdown:: New Plots
```
Suggest removing any with "None" to better match the other METplus components. In Slack on 3/25, I sent the following note: 

> As you know, I always put in a title "Bugfixes", "Enhancements", etc, even if there are none and I put "None" underneath.  I see in the MET and METplus repos they just don't put the title "Bugfixes", "Enhancements", etc, if that category had no changes.  I guess I'll start doing the same for consistency...

bikegeek replied:
This is inconsistent with how thing were originally done.  Initially, when there was nothing, the pull down was left empty.  Then, the practice was to add 'None' to the pull down.

jprestop replied:
Yes, I remember.  It looks like now METplus and MET are simply not including the title if changes for that title are not included, so I followed suit for the METcalcpy and METdataio releases and thought METplotpy and METviewer should do the same for consistency.  Please feel free to add something to the meeting agenda for tomorrow if you'd like to discuss it.  I don't have any strong preferences, except a preference for consistency.

jprestop commented on docs/Users_Guide/release-notes.rst:
```diff
@@ -22,23 +27,35 @@ METplotpy Version 3.1.0-beta2 release notes (20250331)
-   
+
 METplotpy Version 3.1.0-beta1 release notes (20250123)
 ------------------------------------------------------
```
```suggestion
======================================================
```

bikegeek committed:
Update docs/Users_Guide/release-notes.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/release-notes.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/release-notes.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

jprestop commented:
Actually, it looks like METplus IS including the title and None, so I will add those back in for METcalcpy and METdataio.  Sorry for the confusion.

jprestop committed:
Added note directive to upgrade instructions

jprestop reviewed (approved):
I approve this request
